### PR TITLE
feat: Add "longFieldMaxLength" config var

### PR DESCRIPTION
**Pull request by trentm**
https://github.com/elastic/apm/pull/493 specifies a `long_field_max_length` config option we can use for `span.context.db.statement`. Currently this is hardcoded to 10000... bytes. 

- We'll also change to be 10000 unicode characters (rather than bytes). This is in line with the json-schema spec for "maxLength" and is now [the apm spec for `long_field_max_length`](https://github.com/elastic/apm/blob/master/specs/agents/field-limits.md#long_field_max_length-configuration). See [the http-client PR](https://github.com/elastic/apm-nodejs-http-client/pull/162) for this part of the change.
- This also deprecates the existing node.js-agent-only `errorMessageMaxLength` config option. It now defaults to the `longFieldMaxLength` value if not specified. Otherwise `errorMessageMaxLength` is still supported. (We should drop it in the "next major".)
- This change also applies the truncation to the other fields mentioned in [the `long_field_max_length` spec](https://github.com/elastic/apm/blob/master/specs/agents/field-limits.md#long_field_max_length-configuration). This *does* mean that some existing usage out there might get truncation of some of these fields where they didn't before. That's debatable whether that constitutes a "breaking change". So far this PR takes the point of view that this is *not* a breaking change that would require either a *separate* config var (which means not conforming with the APM spec link above) or only applying truncation to these fields in a new major version later. Objections?

Fixes: #1921

### Checklist

- [x] Implement code
- [x] update package.json dep once https://github.com/elastic/apm-nodejs-http-client/pull/162 is merged and v10.0.0 is published
- [x] Add tests
- [x] Update TypeScript typings
- [x] Update documentation
- [x] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by trentm** [comment]
https://github.com/elastic/apm/issues/488 is a discussion for a cross-agent config option name for this var. Let's hold until that discussion plays out.

**Comment by trentm** [comment]
I've started https://github.com/elastic/apm/pull/493 for the spec addition for this field name: `long_field_max_length`.

**Comment by trentm** [comment]
^^ Integration Tests failure is the current https://github.com/elastic/apm-integration-testing/issues/1188 flake

**Comment by trentm** [comment]
Here is a small test script to demonstrate `longFieldMaxLength` working on `span.context.db.statement`:

```
const apm = require('./').start({
  longFieldMaxLength: 100000
})

const { Client } = require('@elastic/elasticsearch')
var client = new Client({
  node: 'http://localhost:9200',
  auth: { username: 'admin', password: 'changeme' }
})

const t1 = apm.startTransaction('t1')
// Build up a search body >10k in size.
aggs = {}
for (var i = 0; i < 1000; i++) {
  aggs[`aggs${i}`] = { "terms": { "field": "author" } }
}
client.search(
    {
      body: {
        size: 1,
        aggregations: aggs
      }
    },
    function (err, result) {
      t1.end()
    }
)
```

File: docs/configuration.asciidoc
Comment (astorm) [inline docs/configuration.asciidoc:686]: nit: As written, I'm a little uncertain what this extra message is trying to convey about the behavior of `errorMessageMaxLength`, and how the bytes configured will be converted into characters.

File: package.json
Comment (astorm) [inline package.json:90]: I presume the intention is to land this PR after the client changes land and that the github repo and `XXX` will be swapped out.

**Review by astorm — COMMENTED** [review]
Overall looks it does the thing the way the agent wants the thing done. Will approve once the `package.json` is updated to reflect what we want to ship.

File: docs/configuration.asciidoc
Comment (trentm) [reply docs/configuration.asciidoc:686]: Given the field is now deprecated, should I just drop that comment to avoid adding confusion?

What is happening is that we accept a config value that is "some number of bytes" (because the suffixes are "b", "kb", etc.). However that is just converted to an integer: `"2kb"` becomes `2048`. Then that `2048` is used as a count of *characters* (not bytes) for translation. It is a little subtle, but almost certainly won't affect anyone -- only if they were having error messages near the limit and the error messages had lots of emoji or chars outside the BMP in them.

File: package.json
Comment (trentm) [reply package.json]: Yup, done.

File: docs/configuration.asciidoc
Comment (astorm) [reply docs/configuration.asciidoc:686]: Ah -- I see, so the string is converted into a byte value, and then that byte value is used as a character length.  I'm not sure whether it's better to say nothing or keep what we have to indicate that's it's not a true limiting by byte size.  Nit withdrawn.

**Review by astorm — APPROVED** [review]

**Pull request merged**
